libs/utils/include/utils/linux/Condition.h

```diff
@@ -68,7 +68,7 @@ class Condition {
     std::cv_status wait_until(std::unique_lock<Mutex>& lock,
             const std::chrono::time_point<std::chrono::steady_clock, D>& timeout_time) noexcept {
         // convert to nanoseconds
-        int64_t ns = std::chrono::duration<int64_t, std::nano>(timeout_time.time_since_epoch()).count();
+        uint64_t ns = std::chrono::duration<uint64_t, std::nano>(timeout_time.time_since_epoch()).count();
```

Comment:
What's the type of `D` here? Is it `uint64_t`? Could we use `D` instead of hard-coding `uint64_t`?

Reply:
Ohhhhh let me give that a try and make sure the test still passes.

Reply:
D is a duration but replacing `uint64_t` with `typename D::rep` might work.

Reply:
Hm, it fails if I use `D::rep` so I'll just stick to uint64_t

Reply:
Sure, was worth a shot. Yay templates